### PR TITLE
feat: implement URL strategies for enqueueLinks

### LINKED ISSUE #29: Add `sameDomain` option to `enqueueLinks` helper
After #28 is done, please add a new option to `enqueueLinks`: `sameDomain: boolean` that will default to `true`. Currently to achieve this, we need pseudourls which are quite verbose and weird, given this use case is probably the most common, we want to support it by default without any configuration.

Old API:

```js
const requestQueue = await Apify.openRequestQueue();
await requestQueue.addRequest({ url: 'https://www.iana.org/' });

const crawler = new Apify.PuppeteerCrawler({
    requestQueue,
    handlePageFunction: async ({ request, page, enqueueLinks }) => {
        const title = await page.title();
        console.log(`Title of ${request.url}: ${title}`);
        await enqueueLinks({
            pseudoUrls: ['https://www.iana.org/[.*]'],
        });
    },
});
```

New API:

```js
const requestQueue = await Apify.openRequestQueue();
await requestQueue.addRequest({ url: 'https://www.iana.org/' });

const crawler = new Apify.PuppeteerCrawler({
    requestQueue,
    handlePageFunction: async ({ request, page, enqueueLinks }) => {
        const title = await page.title();
        console.log(`Title of ${request.url}: ${title}`);
        await enqueueLinks(); // enqueue all links from the same domain, no parameters needed for this use case
    },
});
```

### PR DESCRIPTION
**Pull request by vladfrangu**
Closes #29

### CODE REVIEW

File: packages/core/src/enqueue_links/enqueue_links.ts [hunk 1]
```diff
@@ -63,6 +63,17 @@ export interface EnqueueLinksOptions {
      * ```
      */
     transformRequestFunction?: RequestTransform;
+
+    /**
+     * The strategy to use when enqueueing the urls.
+     * @default DomainStrategy.SameDomain
+     */
+    strategy?: DomainStrategy | 'ALL' | 'SAME_DOMAIN';
```
[the remark this answers is not in the document]
Comment (vladfrangu) [reply packages/core/src/enqueue_links/enqueue_links.ts]: Just noticed that too..

I also tried it without the extra strings but TS wouldn't allow it, so we sadly have to do it this way

[the remark this answers is not in the document]
Comment (B4nan) [reply packages/core/src/enqueue_links/enqueue_links.ts]: I believe we can do `DomainStrategy | keyof typeof DomainStrategy`, using this in MikroORM for the same use case

File: packages/core/src/enqueue_links/enqueue_links.ts [hunk 2]
```diff
@@ -117,11 +128,30 @@ export async function enqueueLinks(options: EnqueueLinksOptions): Promise<QueueO
             ow.object.validate(validators.pseudoUrl),
         ))),
         transformRequestFunction: ow.optional.function,
+        strategy: ow.optional.string.oneOf([
+            DomainStrategy.All,
```
Comment (B4nan) [inline packages/core/src/enqueue_links/enqueue_links.ts:132]: maybe it would make more sense to call it `EnqueueStrategy`?

**Review by B4nan — APPROVED** [review]
otherwise lgtm

File: packages/core/src/enqueue_links/enqueue_links.ts (hunk 2 above)
Comment (vladfrangu) [reply packages/core/src/enqueue_links/enqueue_links.ts:132]: Agreed, will change 👍

**Pull request merged**